api/src/models/service_config/enviroment.rs

```diff
@@ -0,0 +1,372 @@
+                original_value: None,
+            },
+            None => self.clone(),
+        }
+    }
+}
+
+impl Hash for EnvironmentVariable {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.key.hash(state);
+        self.value.unsecure().hash(state);
+    }
+}
+
+impl TryFrom<(String, Value)> for EnvironmentVariable {
+    type Error = &'static str;
+
+    fn try_from(value: (String, Value)) -> Result<Self, Self::Error> {
+        let (key, value) = value;
+
+        let (value, templated, replicate) = match value {
+            Value::String(v) => (SecUtf8::from(v), false, false),
+            Value::Object(values) => {
+                let value = values
+                    .get("value")
+                    .ok_or("Invalid env value payload: value is a required field.")?;
+
+                let value = match value {
+                    Value::String(v) => v,
+                    _ => return Err("Invalid env value payload: value must be a string."),
+                };
+
+                (
+                    SecUtf8::from(value),
+                    values
+                        .get("templated")
+                        .map_or(false, |templated| templated.as_bool().unwrap_or(false)),
+                    values
+                        .get("replicate")
+                        .map_or(false, |replicate| replicate.as_bool().unwrap_or(false)),
+                )
+            }
+            _ => {
+                return Err("Invalid env value payload: The value must be a string or an object.");
+            }
+        };
+
+        Ok(EnvironmentVariable {
+            key,
+            value,
+            original_value: None,
+            templated,
+            replicate,
+        })
+    }
+}
+
+impl PartialEq for EnvironmentVariable {
+    fn eq(&self, other: &Self) -> bool {
+        self.key == other.key && self.value == other.value
+    }
+}
+impl Eq for EnvironmentVariable {}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_json::from_value;
+
+    #[test]
+    fn should_parse_env_from_kv_string() {
+        let e = from_value::<Environment>(serde_json::json!(["MYSQL_USER=admin"]))
+            .unwrap()
+            .values
+            .into_iter()
+            .next()
+            .unwrap();
+
+        assert_eq!(e.key, "MYSQL_USER".to_string());
+        assert_eq!(e.value.unsecure(), "admin".to_string());
+    }
+
+    #[test]
+    fn should_parse_env_from_kv_object() {
+        let e = from_value::<Environment>(serde_json::json!({"MYSQL_USER": "admin"}))
+            .unwrap()
+            .values
+            .into_iter()
+            .next()
+            .unwrap();
+
+        assert_eq!(e.key, "MYSQL_USER".to_string());
+        assert_eq!(e.value.unsecure(), "admin".to_string());
+    }
+
+    #[test]
+    fn should_parse_env_from_object_without_templating() {
+        let e = from_value::<Environment>(serde_json::json!({
+            "MYSQL_USER": {"value": "admin"}
```

Comment:
Here, I have the same question as above: Do you actually verify that templating was not applied?

Reply:
This test is just about parsing only. It just tests if the JSON is valid and if the deserialized value has the correct content.

Reply:
Okay, that makes sense.